platform/posix/transport/src/transport_reconnect_posix.c

```diff
@@ -20,50 +20,55 @@
  */
 
 /**
- * @file reconnect_posix.c
+ * @file transport_reconnect_posix.c
  * @brief Implementation of the backoff logic when connection fails to the server fails.
  */
 
 /* Standard includes. */
 #include <unistd.h>
 #include <stdlib.h>
 #include <time.h>
-#include "reconnect.h"
+#include <math.h>
+#include "transport_reconnect.h"
 
 /*-----------------------------------------------------------*/
 
 bool Transport_ReconnectBackoffAndSleep( TransportReconnectParams_t * pReconnectParams )
 {
     bool status = false;
-    uint32_t jitter = 0;
+    uint32_t nextBackOffAndJitter = 0;
+    uint32_t nextMaxBackOff = 0;
+    uint32_t jitterUpperLimit = MAX_RECONNECT_BACKOFF_SECONDS;
 
     /* If MAX_RECONNECT_ATTEMPTS is set to 0, try forever */
-    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPS ) ||
-        ( 0 == MAX_RECONNECT_ATTEMPS ) )
+    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPTS ) ||
+        ( 0 == MAX_RECONNECT_ATTEMPTS ) )
     {
         /*  Wait for timer to expire for the next reconnect */
-        ( void ) sleep( pReconnectParams->reconnectTimeoutSec );
-
-        /* Calculate the next timeout value only if timeout value has not 
-         * exceeded  MAX_RECONNECT_TIMEOUT_SECONDS */
-        if( pReconnectParams->reconnectTimeoutSec < MAX_RECONNECT_TIMEOUT_SECONDS )
-        {
-            /* Calculate jitter value picking a random number 
-             * between 0 and  MAX_JITTER_VALUE_SECONDS. */
-            jitter = ( rand() % MAX_JITTER_VALUE_SECONDS );
-             /* Double the timeout value for the next iteration */
-            pReconnectParams->reconnectTimeoutSec += pReconnectParams->reconnectTimeoutSec;
-            pReconnectParams->reconnectTimeoutSec += jitter;
-        }
+        ( void ) sleep( pReconnectParams->nextBackOffSec );
 
+        /* Increment backoff counts. */
         pReconnectParams->attemptsDone++;
+
+        /* Calculate the delay time for the next reconnect attempt. */
+
+        /* Calculate upper limit of range for selecting a random value. */
+        nextMaxBackOff = INITIAL_RECONNECT_BACKOFF_SECONDS *
+                         ( 2 << pReconnectParams->attemptsDone );
+        jitterUpperLimit = ( nextMaxBackOff < MAX_RECONNECT_BACKOFF_SECONDS ) ?
```

Comment:
This will set the jitterUpperLimit always to MAX_RECONNECT_BACKOFF_SECONDS once it reaches MAX_RECONNECT_BACKOFF_SECONDS, there is no reason to keep calculating every time once it has reached MAX_RECONNECT_BACKOFF_SECONDS.

Reply:
Have re-factored the implementation to calculate next jitter window max value by either doubling the previous jitter max or using the cap value